hail/src/main/scala/is/hail/expr/ir/BaseIR.scala

```diff
@@ -1,23 +1,53 @@
 package is.hail.expr.ir
 
 import is.hail.types.BaseType
+import is.hail.utils.StackSafe._
+import is.hail.utils._
 
 abstract class BaseIR {
   def typ: BaseType
 
-  def children: IndexedSeq[BaseIR]
+  protected def childrenSeq: IndexedSeq[BaseIR]
 
-  def copy(newChildren: IndexedSeq[BaseIR]): BaseIR
+  def children: Iterable[BaseIR] = childrenSeq
```

Comment:
It's not clear to me why we need `children` and `childrenSeq` - can you explain?

Reply:
It's just separating implementation from interface: the implementation is an IndexedSeq of children, but this makes that fact protected, and only exposes an Iterable interface over children to external clients.

Reply:
Ah ok - I publicised and enforced that children should be an `IndexedSeq` on Tim's suggestion because apparently it was a bottleneck. I like the idea of not doing that.

Reply:
I tried to do this in a way that should't affect performance. Keep an eye out for anything that might.